src/Interpreters/InterpreterSelectQuery.cpp

```diff
@@ -3604,10 +3604,12 @@ void InterpreterSelectQuery::handleSeekToSetting()
         if (seek_points.size() != 1)
             throw Exception(ErrorCodes::NOT_IMPLEMENTED, "It doesn't support time based `seek_to` settings for multiple shards");
 
-        /// Here we rewrite WHERE predicates of SELECT query.
+        /// If the storage does not have built-in seek_to support (for example, external streams like Kafka support seek_to, via offset, natively)
+        /// we rewrite WHERE predicates of SELECT query.
         /// Example : SELECT * FROM stream SETTINGS seek_to='2022-01-01 00:01:01' =>
         /// SELECT * FROM stream WHERE _tp_time >= '2022-01-01 00:01:01'
-        addEventTimePredicate(getSelectQuery(), seek_points[0]);
+        if (storage && !storage->builtinSeekToSupport())
+            addEventTimePredicate(getSelectQuery(), seek_points[0]);
```

Comment:
1) `seek_to` a timestamp is translated a seek and a filtering. We may need do same thing for external stream.
2) StorageStream support seek to by timestamp as well ?

Reply:
@yl-lisen thoughts ?

Reply:
For 1, it depends, if the storage does not support seek to a timestamp, it might need a filter (sometimes, it might not even possible). But for a storage like Kafka, there is no need to do another seek and filter, it's just wasting resources.

Reply:
So, if later on, we found a external stream that needs to apply the filtering, we can't simply use `storage->builtinSeekToSupport`, we will need to defer it to the storage type level.

Reply:
For seek to by timestamp, the event time predicates is added since nativelog seek to by timestamp is not accurate.
for `kafkaSource`, we called `rd_kafka_offsets_for_times`, it also documented:
```
 * @brief Look up the offsets for the given partitions by timestamp.
 *
 * The returned offset for each partition is the earliest offset whose
 * timestamp is greater than or equal to the given timestamp in the
 * corresponding partition.
```
So we can not get an accurate offset from any source whatever is NativeLog or Kafka
@zliang-min

Reply:
Based on above description, it may indeed be accurate for  kafka source

Reply:
I think the method name `supportsNativeSeekTo` is not good, we can use `supportAccurateSeekTo` or something else

Reply:
Thanks @yl-lisen for the context of why the filter was adding at the first place. Based on that, I think `supportsAccurateSeekTo` is a better name. Updated.

Reply:
Kafka timestamp based seek is NOT accurate, it never does. So we shall filter further. It is not waste of resources.